### ISSUE TITLE
Twitter Login: "Failed to request resource." (Possibly OAuth Library?)

### ISSUE THREAD

**Issue by ghost** [definitely246/social#15]
_Not sure if this is a duplicate issue of the other ones pertaining to twitter, but they were a bit vague._

I'm currently getting a `TokenResponseException` with the message of `Failed to request resource.`.

My configuration file is correct, as are the twitter authentication credentials (key and secret) - I've gone as far as refreshing the twitter credentials and updating them in the configuration file.

The exception is being thrown at `Line 68` in `/vendor/lusitanian/oauth/src/OAuth/Common/Http/Client/StreamClient.php`. Digging into it, it appears that the OAuth library is failing when it attempts to get the OAuth token. 

Up until the call to `file_get_contents()` in that file everything appears to look correct; so it could perhaps be an issue with the OAuth library? It's just I can use Facebook authentication fine.

**Comment by ghost** [comment]
Ok, Just realised `file_get_contents()` returns false for 401 response codes. I've dumped out the request information and tried it manually, and I get a 401 response code with a body of "_Failed to validate oauth signature and token_".

This is similar to #12, but occurs before any permission has been granted. (In fact, it occurs when calling (`Social::login('twitter');` - when the URL used for authentication is being requested. )

**Comment by ghost** [comment]
**This was fixed by creating a new twitter app with a placeholder in the callback URL field.**

I had forgot to do this with the first app, and it appeared that adding the placeholder later on didn't work.

**Comment by jrean** [comment]
@fmorrow what do you mean by "This was fixed by creating a new twitter app with a placeholder in the callback URL field." Can you please provide an example?

**Comment by ghost** [comment]
@jrean - Apologies for the delayed response, I totally missed this completely.

I logged in to the Twitter API developer console, and proceeded to register a new application there. In the field requesting a callback URL I opted for a placeholder URL; i.e a dummy link.

I expect that actually you could modify an existing application and provide a dummy link, and then regenerate the keys. (Log in to the application console area, -> API Keys -> "Regenerate API Keys")

In short, I simply forgot to begin my debugging efforts with checking things at Twitter's end!
